src/libmongoc/src/mongoc/mongoc-topology-private.h

```diff
@@ -141,6 +141,11 @@ typedef struct _mongoc_topology_t {
 
    /* This is overridable for SRV polling tests to mock DNS records. */
    _mongoc_rr_resolver_fn rr_resolver;
+
+   /* valid is false when mongoc_topology_new failed to construct a valid topology.
+    * This could occur if the URI is invalid.
+    * An invalid topology does not monitor servers. */
+   bool valid;
```

Comment:
Note: It would be good if mongoc_topology_new() were extended to set this to a default value, but perhaps that's only my pedanticism talking. UPDATE: lo and behold, mongoc_topology_new() DOES initialize this member (although I guess it's just not visible in delta). +1.